tests/Aspire.Hosting.Azure.Tests/AzureCosmosDBEmulatorFunctionalTests.cs

```diff
@@ -248,6 +249,50 @@ await pipeline.ExecuteAsync(async token =>
 
         DockerUtils.AttemptDeleteDockerVolume(volumeName);
     }
+
+    [Fact]
+    [RequiresDocker]
+    public async Task AddAzureCosmosDB_RunAsEmulator_CreatesDatabase()
+    {
+        var cts = new CancellationTokenSource(TimeSpan.FromMinutes(10));
+
+        using var builder = TestDistributedApplicationBuilder.Create(options => { }, testOutputHelper);
+
+        var databaseName = "db1";
+        var containerName = "container1";
+        var partitionKeyPath = "/id";
+
+        var cosmos = builder.AddAzureCosmosDB("cosmos")
+                            .WithDatabase(databaseName, db => db.Containers.Add(new(containerName, partitionKeyPath)))
+                            .RunAsEmulator();
+
+        using var app = builder.Build();
+
+        await app.StartAsync(cts.Token);
+
+        await app.WaitForTextAsync("Started").WaitAsync(cts.Token);
+
+        var rns = app.Services.GetRequiredService<ResourceNotificationService>();
+        await rns.WaitForResourceHealthyAsync(cosmos.Resource.Name, cts.Token);
+
+        var hb = Host.CreateApplicationBuilder();
+        hb.Configuration[$"ConnectionStrings:{cosmos.Resource.Name}"] = await cosmos.Resource.ConnectionStringExpression.GetValueAsync(default);
+        hb.AddAzureCosmosClient(cosmos.Resource.Name);
+
+        using var host = hb.Build();
+
+        await host.StartAsync(cts.Token);
+
+        // This needs to be outside the pipeline because when the CosmosClient is disposed,
+        // there is an exception in the pipeline
+        using var cosmosClient = host.Services.GetRequiredService<CosmosClient>();
+
+        var database = cosmosClient.GetDatabase(databaseName);
+        var container = database.GetContainer(containerName);
+
+        Assert.NotNull(database);
+        Assert.NotNull(container);
```

Comment:
Is this actually making a connection to the Cosmos DB emulator? There is not `await` here, so I assume this will pass whether the db and container are created or not.